### PR TITLE
Set proper timeout for canary workflows

### LINKED ISSUE #394: canary not running properly on v0.23.0
## Expected Behavior

We have been working on rolling out a temporal cluster (with mysql persistence) including having the canary continuously running to help ensure correctness. This seemed to work pretty well on the prior release v0.21.1 but the canary is not working properly on v0.23.0. I'll try to provide as much information here as I can.

To start with, the upgrade to v0.23.0 required me to re-create the temporal database (there was no upgrade path). So the problems I am encountering happened with an empty database.

Once the cluster was upgraded, I started running the canary (with visibility, batch, cancellation, and searchAttributes excluded because these were not working on v0.21.1). It ran for almost 20 minutes then the cron workflow (that runs the sanity workflow that runs the canaries) stopped starting. The canary process did not output any logs, and restarting the process didn't help. This is a graph of the `start` metric from the canary:

image

While the canary was running, the `reset` canary was failing every time (I don't think this happened in the previous version).

Once thing that seems unusual is how the cron workflow is behaving -- it appears to be getting run many times a second yet the sanity workflow never gets started:

image

Other things I noticed, the `timer_tasks` table seems to be growing at more than 1 row per second:

```
mysql> select count(*) from timer_tasks;
+----------+
| count(*) |
+----------+
|   154783 |
+----------+
1 row in set (0.03 sec)
```

Here is the history of one of the many cron workflows:
```
# tctl --ns canary wf show -w cadence.canary.cron -r d44d5b65-3c95-4f74-b319-f4640e8060e8
  1  WorkflowExecutionStarted         {WorkflowType:{Name:cronWorkflow}, ParentInitiatedEventId:0,
                                       TaskList:{Name:canary-task-queue, Kind:Normal},
                                       Input:{"canary","workflow.sanity"},
                                       WorkflowExecutionTimeoutSeconds:1080,
                                       WorkflowRunTimeoutSeconds:1080,
                                       WorkflowTaskTimeoutSeconds:10,
                                       ContinuedExecutionRunId:e12a8224-44ad-4db9-b4ad-4894d484a336,
                                       Initiator:CronSchedule,
                                       ContinuedFailureReason:temporalInternal:Timeout StartToClose,
                                       ContinuedFailureDetails:{}, LastCompletionResult:{},
                                       OriginalExecutionRunId:d44d5b65-3c95-4f74-b319-f4640e8060e8,
                                       FirstExecutionRunId:4178af07-f8b4-4cfe-85f3-7af8bdb64904,
                                       Attempt:0,
                                       WorkflowExecutionExpirationTimestamp:1589853344158000000,
                                       CronSchedule:@every 30s,
                                       FirstDecisionTaskBackoffSeconds:2172214,
                                       PrevAutoResetPoints:{Points:[len=1]}, Header:{Fields:map{}}}
  2  WorkflowExecutionContinuedAsNew  {NewExecutionRunId:9c26d2d2-3256-4dba-9eac-7147a215735b,
                                       WorkflowType:{Name:cronWorkflow},
                                       TaskList:{Name:canary-task-queue, Kind:Normal},
                                       Input:{"canary","workflow.sanity"},
                                       WorkflowRunTimeoutSeconds:1080,
                                       WorkflowTaskTimeoutSeconds:10,
                                       DecisionTaskCompletedEventId:-23,
                                       BackoffStartIntervalInSeconds:2172243,
                                       Initiator:CronSchedule,
                                       FailureReason:temporalInternal:Timeout StartToClose,
                                       FailureDetails:{}, LastCompletionResult:{},
                                       Header:{Fields:map{}}}
```

I'm happy to provide more information, I'm not sure what else I can add here. The workflow is not getting executed so I don't see much in the way of log lines or metics to report.

## Specifications

  - Version: Temporal v0.23.0 with mysql persistence
  - Platform: Linux (k8s)

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
`WorkflowRunTimeout` should be set instead of `WorkflowExecutionTimeout` in canary.

**Why?**
This PR closes #394.

**How did you test it?**
Run canary locally for 1 hour and check workflow state with:
```
./tctl --ns canary wf show -w cadence.canary.cron
```

**Potential risks**
No risks.

### CODE REVIEW

File: canary/sanity.go [hunk 1]
```diff
@@ -42,8 +42,8 @@ var sanityChildWFList = []string{
 	wfTypeCancellation,
 	wfTypeRetry,
 	wfTypeReset,
-	wfTypeHistoryArchival,
-	wfTypeVisibilityArchival,
+	//wfTypeHistoryArchival,
```
Comment (samarabbas) [inline canary/sanity.go:45]: why are you commenting these out?  These can be excluded through config.

Comment (alexshtin) [reply canary/sanity.go:45]: They are not registered and generate errors. Should I register them instead?

Comment (skrul) [reply canary/sanity.go:45]: FWIW I see errors from `wfTypeVisibilityArchival` even after I've added it to the excludes list.

**Comment by skrul** [comment]
I patched this change into my v0.23.0 canary and it seems to have solved the problem 😄

**Comment by skrul** [comment]
Would it be possible to get this merged for the next release?  😄

File: canary/sanity.go (hunk 1 above)
Comment (skrul) [reply canary/sanity.go:45]: I now understand that these will fail if the cluster does not have archiving enabled 😄

Comment (samarabbas) [reply canary/sanity.go:45]: Yes register them instead of commenting it out.  There is already config based mechanism to exclude them.

**Pull request merged**